core/trino-main/src/main/java/io/trino/cost/FilterProjectAggregationStatsRule.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.cost;
+
+import io.trino.Session;
+import io.trino.matching.Pattern;
+import io.trino.sql.planner.TypeProvider;
+import io.trino.sql.planner.iterative.GroupReference;
+import io.trino.sql.planner.iterative.Lookup;
+import io.trino.sql.planner.plan.AggregationNode;
+import io.trino.sql.planner.plan.FilterNode;
+import io.trino.sql.planner.plan.PlanNode;
+import io.trino.sql.planner.plan.ProjectNode;
+
+import java.util.Optional;
+
+import static com.google.common.collect.MoreCollectors.onlyElement;
+import static io.trino.SystemSessionProperties.isUnestimatedFilterCostApproximationEnabled;
+import static io.trino.cost.FilterStatsCalculator.UNKNOWN_FILTER_COEFFICIENT;
+import static io.trino.sql.planner.plan.Patterns.filter;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * AggregationStatsRule does not have sufficient information to provide accurate estimates on aggregated symbols.
+ * This rule finds unestimated filters on top of aggregations whose output row count was estimated and returns
+ * estimated row count of (0.9 * aggregation output row count) for the filter.
+ */
+public class FilterProjectAggregationStatsRule
+        extends SimpleStatsRule<FilterNode>
+{
+    private static final Pattern<FilterNode> PATTERN = filter();
```

Comment:
please try to wrap it in patter as in `io.trino.sql.planner.iterative.rule.AddExchangesBelowPartialAggregationOverGroupIdRuleSet`.
It will be easier to move to exhaustive CBO in the future then